apps/emqx_bridge_datalayers/test/emqx_bridge_datalayers_arrow_SUITE.erl

```diff
@@ -0,0 +1,461 @@
+    Name = atom_to_binary(?MODULE),
+    {ConnConfString, ConnConfMap} = datalayers_connector_config(
+        Name, DatalayersHost, DatalayersPort, Config
+    ),
+    {ActionConfString, ActionConfMap} = datalayers_action_config(Name, Config),
+    {Name, ConnConfString, ConnConfMap, ActionConfString, ActionConfMap}.
+
+datalayers_connector_config(Name, DatalayersHost, DatalayersPort, Config) ->
+    UseTLS = proplists:get_value(use_tls, Config, false),
+    ConfigString =
+        io_lib:format(
+            "connectors.datalayers.~s {\n"
+            "  enable = true\n"
+            "  server = \"~s:~b\"\n"
+            "  parameters {\n"
+            "    driver_type = arrow_flight\n"
+            "    database = mqtt\n"
+            "    username = admin\n"
+            "    password = public\n"
+            "  }\n"
+            "  ssl {\n"
+            "    enable = ~p\n"
+            "    verify = verify_peer\n"
+            "    cacertfile = \"~s\"\n"
+            "  }\n"
+            "}\n",
+            [
+                Name,
+                DatalayersHost,
+                DatalayersPort,
+                UseTLS,
+                cacert_file()
+            ]
+        ),
+    {ConfigString, parse_and_check_connector(ConfigString, Name)}.
+
+datalayers_action_config(Name, Config) ->
+    BatchSize = proplists:get_value(batch_size, Config, 100),
+    QueryMode = proplists:get_value(query_mode, Config, sync),
+    SQL = sql_template(),
+    ConfigString =
+        io_lib:format(
+            "actions.datalayers.~s {\n"
+            "  enable = true\n"
+            "  connector = ~s\n"
+            "  parameters {\n"
+            "    sql = \"~s\"\n"
+            "  }\n"
+            "  resource_opts = {\n"
+            "    request_ttl = 15s\n"
+            "    query_mode = ~s\n"
+            "    batch_size = ~b\n"
+            "  }\n"
+            "}\n",
+            [
+                Name,
+                Name,
+                SQL,
+                QueryMode,
+                BatchSize
+            ]
+        ),
+    {ConfigString, parse_and_check_action(ConfigString, Name)}.
+
+sql_template() ->
+    <<
+        "INSERT INTO mqtt_arrow.common_test(ts, msgid, clientid, topic, qos, payload) VALUES "
+        "(${timestamp}, ${id}, ${clientid}, ${topic}, ${qos}, ${payload})"
+    >>.
+
+parse_and_check_connector(ConfigString, Name) ->
+    {ok, RawConf} = hocon:binary(ConfigString, #{format => map}),
+    hocon_tconf:check_plain(emqx_connector_schema, RawConf, #{required => false, atom_key => false}),
+    #{<<"connectors">> := #{<<"datalayers">> := #{Name := Config}}} = RawConf,
+    Config.
+
+parse_and_check_action(ConfigString, Name) ->
+    {ok, RawConf} = hocon:binary(ConfigString, #{format => map}),
+    hocon_tconf:check_plain(emqx_bridge_v2_schema, RawConf, #{required => false, atom_key => false}),
+    #{<<"actions">> := #{<<"datalayers">> := #{Name := Config}}} = RawConf,
+    Config.
```

Comment:
Note that the function here does not assert that it processed the correct config root.

Please use `emqx_bridge_v2_testlib:parse_and_check`.